ChangeLog

```diff
@@ -9,6 +9,29 @@ Release date: TBA
 ..
   Put new features here and also in 'doc/whatsnew/2.13.rst'
 
+* Added several checkers to deal with unicode security issues
+  (see `Trojan Sources <https://trojansource.codes/>`_ and
+  `PEP 672 <https://www.python.org/dev/peps/pep-0672/>`_ for details) that also
+  concern the readability of the code. In detail the following checks were added:
+
+  * ``invalid-unicode-codec`` checks that UTF-16 and UTF-32 is not used to encode the
+    file. These encodings are currently not valid for Python but could be in the future.
+    As this could break the other checks, we want to make users aware of the issue.
```

Comment:
Working now a bit with UTF-16/32 compatibility I still think this could be an issue for end users.
Normally People don't care what encoding their file is.
UTF-16 and UTF-32 introduce a number of issues (i.e. streaming with line endings does not work, because if you split for `b"\n"` you miss the zero required by UTF-16/32.

If Python ever supports UTF-16/32 end users will most likely don't hear the news. This could lead to unexpected behaviour.
Use case: Imagine Python 3.15 starts to support UTF-16 and a user with an Asian Language Background (for whom UTF-16 makes sense) set their default editor to it. This would lead to strange encoding errors for people using Python 3.14.

Also even so I tried to add tests for UTF-16/32 it still is hard to verify that it will work as expected as long Python doesn't support it.
So I see it as a valid note for end users.

On thought so: I could add a check that the encoding is  UTF-8 as this is the recommended encoding for Python.
If they aren't UTF-8 (and not UTF-16/32), I can raise a `wrong-encoding` convention warning. So this checker would have a bit more sense to end users and actually does something that is according to [PEP8](https://www.python.org/dev/peps/pep-0008/#id20)

Does this sound better to you?

Reply:
@DanielNoord Implemented it and changed the ChangeLog

Reply:
This sound very logical to me and makes a lot sense! Thanks @CarliJoy!